core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go

```diff
@@ -101,7 +101,13 @@ func (builtin *AddServiceCapabilities) Interpret(arguments *builtin_argument.Arg
 		return nil, interpretationErr
 	}
 
-	builtin.serviceName = service.ServiceName(serviceName.GoString())
+	serviceNameGoStr := serviceName.GoString()
+
+	if isValidServiceName := service.IsServiceNameValid(serviceNameGoStr); !isValidServiceName {
+		return nil, startosis_errors.NewInterpretationError("Service name '%v' is invalid as it contains disallowed characters. Service names can only contain characters 'a-z', 'A-Z', '0-9', '-' & '_'", serviceNameGoStr)
+	}
```

Comment:
Should we push this validation into the ServiceConfig object instead, as a field validator? This way we don't have to duplicate this bit in both add_service and add_services. Wgyt?

Reply:
The ServiceConfig object doesn't hold a service_name. Am I interpreting this incorrectly?

Reply:
Have added this to the validator in `add_service_shared` as discussed on Slack!